emscripten/native/network/HTTPClientEmscriptenImpl.cpp

```diff
@@ -0,0 +1,45 @@
+#include "HTTPClientEmscriptenImpl.h"
+
+#include <stdio.h>
+#include <string.h>
+#include <math.h>
+#include <iostream>
+#include <emscripten/fetch.h>
+
+namespace carto {
+    HTTPClient::EmscriptenImpl::EmscriptenImpl(bool log) :
+        _log(log),
+        _timeout(-1)
+    {
+    }
+
+    void HTTPClient::EmscriptenImpl::setTimeout(int milliseconds) {
+        _timeout = milliseconds;
+    }
+
+    bool HTTPClient::EmscriptenImpl::makeRequest(const HTTPClient::Request& request, HeadersFunc headersFn, DataFunc dataFn) const {
+        emscripten_fetch_attr_t attr;
+        emscripten_fetch_attr_init(&attr);
+        strcpy(attr.requestMethod, "GET");
```

Comment:
I believe `request.method` should be used here instead of "GET".

Reply:
Done. I also added headers to request.